src/screens/groups/components/batch-recipients-list/batch-recipients-list.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import { Checkbox, Table } from 'semantic-ui-react'
+import type { RecipientDTO } from '@/lib/dto/RecipientDTO'
+import type { SelectedRecipients } from '@/screens/groups/batch-payment-screen'
+import BatchRecipientInfo from './batch-recipient-info'
+
+interface Props {
+  recipients: RecipientDTO[]
+  selectedRecipients: SelectedRecipients
+  onRowSelect: (_id: string) => void
+  onAmountChange: (_id: string, _amount: number) => void
+  onSelectAll: () => void
+  isSelectAllChecked: boolean
+}
+
+const Component = ({
+  recipients,
+  selectedRecipients,
+  onRowSelect,
+  onAmountChange,
+  onSelectAll,
+  isSelectAllChecked,
+}: Props) => {
+  const handleRowSelect = (id: string) => {
+    const selectedAmount = selectedRecipients[id]?.amount
+    if (selectedAmount === 0) return
+    onRowSelect(id)
+  }
+
+  const handleAmountChange = (
+    id: string,
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const amount = event.target.value ? parseFloat(event.target.value) : null
```

Comment:
I think there is another possible approach how to simplify this line:

```
const amount = parseFloat(event.target.value)
onAmountChange(id, finalAmount)
```

We don't need to create `amount` variable with `null` value and in the next line check this variable for null-ish value. So, I think we can simplify code with an example from above.

Reply:
we need to set the value to 0 if `parseFloat(event.target.value)` is NaN
`const amount = parseFloat(event.target.value) || 0`
`onAmountChange(id, amount)`